### PR TITLE
Preserve dtypes when get() is called with an index

### LINKED ISSUE #324: CategoricalDtype not preserved when using index in Table.get()
When indexing a table that contains a scheme with labels with an index when getting it with `get(index)` we are loosing the information about allowed categories as it returns no longer a `CategoricalDtype` object. This can also break your code as the column can no longer be accessed via `.cat` which is possible without using an index.

This is related to https://github.com/audeering/audformat/issues/317 as we also loose this information when using `map`.

```python
import audformat


db = audformat.Database('db')

db.schemes['spk'] = audformat.Scheme('str', labels=['a', 'b'])
index = audformat.filewise_index(['f1'])
db['files'] = audformat.Table(index)
db['files']['spk'] = audformat.Column(scheme_id='spk')
db['files']['spk'].set(['a'])

db.schemes['label'] = audformat.Scheme('int')
index = audformat.segmented_index(['f1', 'f1'], [0, 1], [1, 2])
db['segments'] = audformat.Table(index)
db['segments']['label'] = audformat.Column(scheme_id='label')
db['segments']['label'].set([0, 1])
```

Now we can do the following:

```python
>>> df = db['files'].get()
>>> df.spk.cat.categories
Index(['a', 'b'], dtype='object')
>>> df.loc['f1', 'spk'] = 'c'
...
TypeError: Cannot setitem on a Categorical with a new category (c), set the categories first
```

But, when indexing by the other table we loose all this information:

```python
>>> df = db['files'].get(db['segments'].index)
>>> df.spk.cat.categories
...
AttributeError: Can only use .cat accessor with a 'category' dtype
>>> df.loc['f1', 'spk'] = 'c'
>>> df
                                     spk
file start           end                
f1   0 days 00:00:00 0 days 00:00:01   c
     0 days 00:00:01 0 days 00:00:02   c
```

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes #324

Solves issue described in #324 and adds a test for it.

### CODE REVIEW

File: audformat/core/table.py [hunk 1]
```diff
@@ -1533,26 +1533,20 @@ def pick_files(
...
             result = self.df.loc[index]
         else:
             files = index.get_level_values(define.IndexField.FILE)
             if self.is_filewise:  # index is segmented
-                result = pd.DataFrame(
-                    self.df.loc[files].values,
-                    index,
-                    columns=self.columns
-                )
-                result_is_copy = True  # to avoid another copy
```
[the remark this answers is not in the document]
Comment (frankenjoe) [reply audformat/core/table.py:1550]: Exactly, and the cause of the bug was actually that - for whatever reasons - we created a new `DataFrame` in that particular case.

**Comment by hagenw** [comment]
Interestingly, this does not solve #324 completely:

```python
import audformat


db = audformat.Database('db')

db.schemes['spk'] = audformat.Scheme('str', labels=['a', 'b'])
index = audformat.filewise_index(['f1'])
db['files'] = audformat.Table(index)
db['files']['spk'] = audformat.Column(scheme_id='spk')
db['files']['spk'].set(['a'])

db.schemes['label'] = audformat.Scheme('int')
index = audformat.segmented_index(['f1', 'f1'], [0, 1], [1, 2])
db['segments'] = audformat.Table(index)
db['segments']['label'] = audformat.Column(scheme_id='label')
db['segments']['label'].set([0, 1])
```

and then:

```python
>>> df = db['files'].get()
>>> df.loc['f1', 'spk'] = 'c'
...
TypeError: Cannot setitem on a Categorical with a new category (c), set the categories first
>>> df
     spk
file    
f1     a
```

but

```python
>>> df = db['files'].get(db['segments'].index)
>>> df.loc['f1', 'spk'] = 'c'
>>> df
                                     spk
file start           end                
f1   0 days 00:00:00 0 days 00:00:01   c
     0 days 00:00:01 0 days 00:00:02   c
```

But maybe the difference in behavior is now attributed to the fact that the first is a filewise index whereas the second a segmented index?

**Comment by frankenjoe** [comment]
> But maybe the difference in behavior is now attributed to the fact that the first is a filewise index whereas the second a segmented index?

Maybe a bug in `pandas` as in both cases the dtypes seem to be right:

```python
df = db['files'].get()
df.spk.cat.categories
```
Index(['a', 'b'], dtype='object')
```

```python
df = db['files'].get(db['segments'].index)
df.spk.cat.categories
```
```
Index(['a', 'b'], dtype='object')
```

So I guess there is not much we can do about it.

**Comment by hagenw** [comment]
I would then propose that we remove "Closes #324" here and update the issue after this is merged.
Afterwards we can then have a look if this is reported somewhere in `pandas` already.

**Comment by frankenjoe** [comment]
Ok, but then we should at least remove the 'bug'  label from #324 as I don't see we can do anything else but make sure the categorical dtype is kept.

**Pull request merged**